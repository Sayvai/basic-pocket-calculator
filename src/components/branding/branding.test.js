import React from 'react';

import { render, screen } from '@testing-library/react';
import Branding from './branding';

describe('Branding component', () => {
  let props;

  beforeEach(() => {
    props = {
      name: '',
      assetUrl: '',
      backgroundColorHex: '',
    };
  });

  it(`should render
      - the wrapper with inline style on background color set via props.backgroundColorHex,
      - the image element (with src={props.altUrl}, and alt={props.name logo}),
      - and sub-header`, () => {
    // given
    props.backgroundColorHex = '#00f';
    props.assetUrl = 'https://image.host.com';

    // when
    const { container } = render(<Branding {...props} />);

    // then
    expect(container).toMatchSnapshot();
  });

  it(`should render
      - the wrapper with inline style on background color set via props.backgroundColorHex,
      - the main header element set via props.name,
      - and sub-header`, () => {
    // given
    props.backgroundColorHex = '#0f0';
    props.name = 'Custom Brand';

    // when
    const { container } = render(<Branding {...props} />);

    // then
    expect(container).toMatchSnapshot();
  });
});