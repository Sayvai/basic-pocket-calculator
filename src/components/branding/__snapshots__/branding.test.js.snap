// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Branding component should render
      - the wrapper with inline style on background color set via props.backgroundColorHex,
      - the image element (with src={props.altUrl}, and alt={props.name logo}),
      - and sub-header 1`] = `
<div>
  <div
    class="branding"
    data-testid="component-branding"
    style="background-color: rgb(0, 0, 255);"
  >
    <img
      alt=" logo"
      src="https://image.host.com"
    />
    <h3>
      Pocket Calculator
    </h3>
  </div>
</div>
`;

exports[`Branding component should render
      - the wrapper with inline style on background color set via props.backgroundColorHex,
      - the main header element set via props.name,
      - and sub-header 1`] = `
<div>
  <div
    class="branding"
    data-testid="component-branding"
    style="background-color: rgb(0, 255, 0);"
  >
    <h2>
      Custom Brand
    </h2>
    <h3>
      Pocket Calculator
    </h3>
  </div>
</div>
`;
